src/main/java/com/ncc/aif/AIFUtils.java

```diff
@@ -0,0 +1,700 @@
+    }
+
+    public static Resource markAudioJustification(Model model, Collection<Resource> toMarkOn, String docId,
+                                                  Double startTimestamp, Double endTimestamp,
+                                                  Resource system, Double confidence) {
+        final Resource justification = makeAudioJustification(model, docId, startTimestamp, endTimestamp, system, confidence);
+        markJustification(toMarkOn, justification);
+        return justification;
+    }
+
+    public static Resource markCompoundJustification(Model model, Collection<Resource> toMarkOn,
+                                                     Collection<Resource> justifications,
+                                                     Resource system, Double confidence) {
+        final Resource compoundJustification = makeAIFResource(model, null,
+                AidaAnnotationOntology.COMPOUND_JUSTIFICATION_CLASS, system);
+        markConfidence(model, compoundJustification, confidence, system);
+        justifications.forEach(j -> compoundJustification.addProperty(AidaAnnotationOntology.CONTAINED_JUSTIFICATION, j));
+        markJustification(toMarkOn, compoundJustification);
+        return compoundJustification;
+    }
+
+    /**
+     * Mark a confidence value on a resource.
+     */
+    public static void markConfidence(Model model, Resource toMarkOn, Double confidence, Resource system) {
+        Resource confidenceBlankNode = model.createResource();
+        confidenceBlankNode.addProperty(RDF.type, AidaAnnotationOntology.CONFIDENCE_CLASS);
+        confidenceBlankNode.addProperty(AidaAnnotationOntology.CONFIDENCE_VALUE, model.createTypedLiteral(confidence));
+        markSystem(confidenceBlankNode, system);
+        toMarkOn.addProperty(AidaAnnotationOntology.CONFIDENCE, confidenceBlankNode);
+    }
+
+    /**
+     * Mark the given resources as mutually exclusive.
+     * <p>
+     * This is a special case of [markAsMutuallyExclusive] where the alternatives are each single edges,
+     * so we simply wrap each edge in a collection and pass to markAsMutuallyExclusive.
+     *
+     * @param alternatives       is a map from alternate edges to the confidence associated with each alternative.
+     * @param noneOfTheAboveProb - if non-null, the given confidence will be applied to the
+     *                           "none of the above" option.
+     * @return The mutual exclusion assertion.
+     */
+    public static Resource markEdgesAsMutuallyExclusive(Model model, ImmutableMap<Resource, Double> alternatives,
+                                                        Resource system, Double noneOfTheAboveProb) {
+
+        HashMap<Collection<Resource>, Double> newAltMap = new HashMap<>();
+        alternatives.keySet().forEach(edge -> {
+            Collection<Resource> edges = new HashSet<>();
+            edges.add(edge);
+            newAltMap.put(edges, alternatives.get(edge));
```

Comment:
Saw this checking for other determinism issues - in this case it is harmless, but in general seeing a mutable collection used as a map key is going to raise red flags for readers, so it might be better to use `ImmutableSet.of(edge)` as the key.  

You also lose determinism from the outer `HashMap`, but Jena probably discards the ordering information anyway, so it doesn't matter.

Reply:
Fixed-- this change will be in tonight's commit.  Thanks again.